### PR TITLE
Hessp/node errors

### PR DESCRIPTION
**Pull request by Paul-Hess**

### CODE REVIEW

**Comment by Paul-Hess** [comment]
more to come in this vein, but this is already at twenty files

**Comment by Paul-Hess** [comment]
response in part to https://github.com/flightstats/hub/issues/1041

File: src/test/javascript/integration/bulk_get_spec.js [hunk 1]
```diff
@@ -1,18 +1,27 @@
 require('../integration_config');
-
+const {
+  fromObjectPath,
+  getResponseBody,
+  getStatusCode,
+  getUris,
+} = require('../lib/helpers');
+
+// TODO: let (pun intended) us plan to use const and, uh ... let
```
Comment (Exide) [inline src/test/javascript/integration/bulk_get_spec.js]: Yes please :)

File: src/test/javascript/integration/bulk_get_spec.js [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
-
 /**
  * create a channel
  * post 4 items
  * stream items back with bulk API from all query endpoints, using bulk & batch params
  */
 describe(testName, function () {
-
+    /*
+      TODO: is this an it block ? if it is more than a helper,
```
Comment (Exide) [inline src/test/javascript/integration/bulk_get_spec.js]: Agreed.

File: src/test/javascript/lib/helpers/functional.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// generic functional helpers
```
Comment (Exide) [inline src/test/javascript/lib/helpers/functional.js]: I like these.

File: src/test/javascript/lib/helpers/getters.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// we have several places we are pulling the same exact keys out of response objects
```
Comment (Exide) [inline src/test/javascript/lib/helpers/getters.js]: I feel like this extra abstraction isn't necessary currently. Calling `getProp('statusCode', response)` is just as readable as `getStatusCode(response)`. Same with `getUris` and `getResponseBody`.

Putting `getStatusCode` and `getResponseBody` in an HTTP lib and `getUris` in a Hub lib might make sense in the future.

**Comment by Exide** [comment]
Go ahead and merge after you run it through iad.dev and verify everything still passes.

**Pull request merged**